### PR TITLE
OHRM5X-1378 : Add Reviews

### PR DESCRIPTION
**Pull request by SharukaNP**

### CODE REVIEW

File: src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+          </oxd-grid>
+        </oxd-form-row>
+        <oxd-form-row>
+          <oxd-grid :cols="3" class="orangehrm-full-width-grid">
+            <oxd-grid-item>
+              <supervisor-autocomplete
+                v-model="review.supervisorReviewer"
+                :rules="rules.supervisorReviewer"
+                required
+                :subordinate="review.employee"
```
Comment (Super-Chama) [inline src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue:47]:
```suggestion
                :subordinate="review.employee"
                required
```

File: src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+            </oxd-grid-item>
+          </oxd-grid>
+        </oxd-form-row>
+        <oxd-form-row>
+          <oxd-grid :cols="3" class="orangehrm-full-width-grid">
+            <oxd-grid-item>
+              <supervisor-autocomplete
+                v-model="review.supervisorReviewer"
+                :rules="rules.supervisorReviewer"
+                required
```
Comment (Super-Chama) [inline src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue:46]:
```suggestion
```

File: src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+          <div class="orangehrm-button-row">
+            <oxd-button
+              display-type="ghost"
+              :label="$t('general.cancel')"
+              type="button"
+              @click="onCancel"
+            />
+            <oxd-button
+              display-type="ghost"
+              :label="$t('general.nage_reviews')"
```
Comment (Super-Chama) [inline src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue:92]:
```suggestion
              :label="$t('general.manage_reviews')"
```

File: src/client/src/orangehrmPerformancePlugin/pages/review.scss [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+// without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along with this program;
+// if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+// Boston, MA  02110-1301, USA
+
+@import '@ohrm/oxd/styles/_mixins.scss';
+
+/*.footer {
```
Comment (Super-Chama) [inline src/client/src/orangehrmPerformancePlugin/pages/review.scss:19]: remove commented code

File: src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+              type="button"
+              @click="onSave"
+            />
+            <oxd-button
+              class="orangehrm-left-space"
```
Comment (Super-Chama) [inline src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue:97]: this class is not needed as you've added `gap: 0.4rem;` in `.orangehrm-button-row`

File: src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+            <oxd-button
+              display-type="ghost"
+              :label="$t('general.cancel')"
+              type="button"
+              @click="onCancel"
+            />
+            <oxd-button
+              display-type="ghost"
+              :label="$t('general.nage_reviews')"
+              type="button"
```
Comment (Super-Chama) [inline src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue:93]:
```suggestion
              type="submit"
```

File: src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+        supervisorReviewer: [required],
+        startDate: [
+          required,
+          validDateFormat(),
+          startDateShouldBeBeforeEndDate(
+            () => this.review.endDate,
+            this.$t(
+              'general.review_period_start_date_should_be_before_end_date',
+            ),
+            {allowSameDate: false},
```
Comment (Super-Chama) [inline src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue:159]: `{allowSameDate: false}` this is default i think, no need to add explicitly

File: src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+        })
+        .then(() => {
+          return this.$toast.saveSuccess();
+        })
+        .then(() => {
+          // go back
+          this.onCancel();
+        });
+    },
+    onActivate() {
```
Comment (Super-Chama) [inline src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue:208]: what if we take both `onActivate()` and `onSave()` to one function?

```
onSave(activate = false) {
  this.isLoading = true;
  this.http
    .create({
      empNumber: this.review.employee.id,
      reviewer: this.review.supervisorReviewer.id,
      startDate: this.review.startDate,
      endDate: this.review.endDate,
      dueDate: this.review.dueDate,
      activate,
    })
    .then(() => {
      return this.$toast.saveSuccess();
    })
    .catch((error) => {
      return this.$toast.warn({
        title: this.$t("general.warning"),
        message: error,
      });
    })
    .then(() => {
      this.onCancel();
    });
}
```
finally in your activate button `@click="onSave(true)"` and for form submit `@submitValid="onSave(false)"`

File: src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+          startDate: this.review.startDate,
+          endDate: this.review.endDate,
+          dueDate: this.review.dueDate,
+          activate: true,
+        })
+        .then(() => {
+          return this.$toast.saveSuccess();
+        })
+        .catch(error => {
+          return this.$toast.warn({
```
Comment (Super-Chama) [inline src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue:223]: Hope you consulted with @devishke-orange about bypassing the 422 triggering the interceptor

File: src/client/src/orangehrmPerformancePlugin/pages/EditReview.vue [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+          <div class="orangehrm-button-row">
+            <oxd-button
+              display-type="ghost"
+              :label="$t('general.cancel')"
+              type="button"
+              @click="onCancel"
+            />
+            <oxd-button
+              display-type="ghost"
+              :label="$t('general.nage_reviews')"
```
Comment (Super-Chama) [inline src/client/src/orangehrmPerformancePlugin/pages/EditReview.vue:94]: see above comments

File: src/client/src/orangehrmPerformancePlugin/pages/EditReview.vue [hunk 11]
```diff
@@ -0,0 +1,260 @@
...
+        .update(this.reviewId, {
+          empNumber: this.review.employee.id,
+          reviewer: this.review.supervisorReviewer.id,
+          startDate: this.review.startDate,
+          endDate: this.review.endDate,
+          dueDate: this.review.dueDate,
+          activate: false,
+        })
+        .then(() => {
+          return this.$toast.saveSuccess();
```
Comment (Super-Chama) [inline src/client/src/orangehrmPerformancePlugin/pages/EditReview.vue:225]: should be `this.$toast.updateSuccess();`

Comment (Super-Chama) [reply src/client/src/orangehrmPerformancePlugin/pages/EditReview.vue:225]: see above comments regarding merging `onSave()` and `onActiviate()`

File: src/client/src/orangehrmPerformancePlugin/pages/review.scss [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+  @include oxd-respond-to('md') {
+    margin-left: unset;
+    max-width: unset;
+    ::v-deep(.oxd-button--medium) {
+      width: unset;
+    }
+  }
+}
+
+.button {
```
Comment (Super-Chama) [inline src/client/src/orangehrmPerformancePlugin/pages/review.scss:51]: do we need this? in above we remove max width which automatically allow the button to grow with container width

**Review by Super-Chama — CHANGES_REQUESTED** [review]
Please run `yarn lint`

File: src/plugins/orangehrmPerformancePlugin/Api/Model/SupervisorModel.php [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Performance\Api\Model;
+
+use OrangeHRM\Core\Api\V2\Serializer\ModelTrait;
+use OrangeHRM\Core\Api\V2\Serializer\Normalizable;
+use OrangeHRM\Entity\ReportTo;
+
+class SupervisorModel implements Normalizable
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Api/Model/SupervisorModel.php:26]: need to discuss this

File: src/plugins/orangehrmPerformancePlugin/Api/PerformanceReviewAPI.php [hunk 14]
```diff
@@ -0,0 +1,238 @@
...
+use OrangeHRM\Performance\Traits\Service\PerformanceReviewServiceTrait;
+use OrangeHRM\Pim\Traits\Service\EmployeeServiceTrait;
+
+class PerformanceReviewAPI extends Endpoint implements CrudEndpoint
+{
+    use PerformanceReviewServiceTrait;
+    use EmployeeServiceTrait;
+    use DateTimeHelperTrait;
+
+    public const PARAMETER_REVIEWER = 'reviewer';
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Api/PerformanceReviewAPI.php:45]:
```suggestion
    public const PARAMETER_REVIEWER_EMP_NUMBER = 'reviewerEmpNumber';
```
:thinking:

File: src/plugins/orangehrmPerformancePlugin/Api/PerformanceReviewAPI.php [hunk 15]
```diff
@@ -0,0 +1,238 @@
...
+        );
+        $performanceReview->setWorkPeriodEnd(
+            $this->getRequestParams()->getDateTime(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_PERIOD_END_DATE)
+        );
+        $performanceReview->setDueDate(
+            $this->getRequestParams()->getDateTime(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_DUE_DATE)
+        );
+        $employee = $this->getEmployeeService()->getEmployeeDao()->getEmployeeByEmpNumber($empNumber);
+        $performanceReview->setJobTitle($employee->getJobTitle());
+        $performanceReview->setDepartment($employee->getSubDivision());
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Api/PerformanceReviewAPI.php:160]:
```suggestion
        $performanceReview->setSubunit($employee->getSubDivision());
```

File: src/plugins/orangehrmPerformancePlugin/Api/PerformanceReviewAPI.php [hunk 16]
```diff
@@ -0,0 +1,238 @@
...
+        if ($this->getRequestParams()->getBooleanOrNull(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_ACTIVATE) == true) {
+            try {
+                $review->setActivatedDate($this->getDateTimeHelper()->getNow());
+                $review->setStatusId(2);
+                $this->getPerformanceReviewService()->updateActivateReview($review);
+            } catch (ReviewServiceException $e) {
+                throw $this->getBadRequestException($e->getMessage());
+            }
+        } else {
+            $review->setStatusId(1);
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Api/PerformanceReviewAPI.php:222]:
```suggestion
            $review->setStatusId(Class::CONST);
```

File: src/plugins/orangehrmPerformancePlugin/Api/PerformanceReviewAPI.php [hunk 17]
```diff
@@ -0,0 +1,238 @@
...
+    public function update(): EndpointResult
+    {
+        $id = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, CommonParams::PARAMETER_ID);
+        $review = $this->getPerformanceReviewService()->getPerformanceReviewDao()->getReviewById($id);
+        $this->throwRecordNotFoundExceptionIfNotExist($review, PerformanceReview::class);
+        $this->setReviewParams($review);
+        if ($this->getRequestParams()->getBooleanOrNull(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_ACTIVATE) == true) {
+            try {
+                $review->setActivatedDate($this->getDateTimeHelper()->getNow());
+                $review->setStatusId(2);
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Api/PerformanceReviewAPI.php:216]:
```suggestion
                $review->setStatusId(Class::CONST);
```

File: src/plugins/orangehrmPerformancePlugin/Api/ReviewEmployeeSupervisorAPI.php [hunk 18]
```diff
@@ -0,0 +1,129 @@
...
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Performance\Api\Model\SupervisorModel;
+use OrangeHRM\Performance\Dto\ReviewEmployeeSupervisorSearchFilterParams;
+use OrangeHRM\Performance\Traits\Service\PerformanceReviewServiceTrait;
+
+class ReviewEmployeeSupervisorAPI extends Endpoint implements CrudEndpoint
+{
+    use PerformanceReviewServiceTrait;
+
+    public const FILTER_EMPLOYEE_NUMBER = 'empNumber';
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Api/ReviewEmployeeSupervisorAPI.php:40]: can use from common params

File: src/plugins/orangehrmPerformancePlugin/Api/ReviewEmployeeSupervisorAPI.php [hunk 19]
```diff
@@ -0,0 +1,129 @@
...
+    public const FILTER_NAME_OR_ID = 'nameOrId';
+    public const PARAM_RULE_FILTER_NAME_OR_ID_MAX_LENGTH = 100;
+
+    public function getAll(): EndpointResult
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Api/ReviewEmployeeSupervisorAPI.php:44]: add doc comments for all methods

File: src/plugins/orangehrmPerformancePlugin/Api/ReviewEmployeeSupervisorAPI.php [hunk 20]
```diff
@@ -0,0 +1,129 @@
...
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Performance\Api\Model\SupervisorModel;
+use OrangeHRM\Performance\Dto\ReviewEmployeeSupervisorSearchFilterParams;
+use OrangeHRM\Performance\Traits\Service\PerformanceReviewServiceTrait;
+
+class ReviewEmployeeSupervisorAPI extends Endpoint implements CrudEndpoint
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Api/ReviewEmployeeSupervisorAPI.php:36]: need to discuss about this API

File: src/plugins/orangehrmPerformancePlugin/Controller/PerformanceReviewSaveController.php [hunk 21]
```diff
@@ -0,0 +1,43 @@
...
+use OrangeHRM\Framework\Http\Request;
+
+class PerformanceReviewSaveController extends AbstractVueController
+{
+    /**
+     * @inheritDoc
+     */
+    public function preRender(Request $request): void
+    {
+        $id = $request->get('id');
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Controller/PerformanceReviewSaveController.php:34]:
```suggestion
        $id = $request->attributes->get('id');
```
> **Warning**
> don't directly use `$request->get`

File: src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php [hunk 22]
```diff
@@ -0,0 +1,122 @@
...
+     */
+    private function saveReviewer(PerformanceReview $performanceReview, string $reviewerGroupName, ?int $reviewerId)
+    {
+        $reviewer = new Reviewer();
+        if (! is_null($reviewerId)) {
+            $reviewer->getDecorator()->setEmployeeByEmpNumber($reviewerId);
+        } else {
+            $reviewer->setEmployee($performanceReview->getEmployee());
+        }
+        $reviewer->setStatus(1);
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php:84]:
```suggestion
        $reviewer->setStatus(Class::CONST);
```

File: src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php [hunk 23]
```diff
@@ -0,0 +1,122 @@
...
+        return $performanceReview;
+    }
+
+    /**
+     * @param PerformanceReview $performanceReview
+     * @param string $reviewerGroupName
+     * @param int|null $reviewerId
+     * @return void
+     */
+    private function saveReviewer(PerformanceReview $performanceReview, string $reviewerGroupName, ?int $reviewerId)
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php:76]:
```suggestion
    private function saveReviewer(PerformanceReview $performanceReview, string $reviewerGroupName, ?int $reviewerEmpNumber)
```
:thinking:

File: src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php [hunk 24]
```diff
@@ -0,0 +1,122 @@
...
+     * @param int $reviewerId
+     * @return PerformanceReview
+     */
+    public function updateReview(PerformanceReview $performanceReview): PerformanceReview
+    {
+        $this->persist($performanceReview);
+        return $performanceReview;
+    }
+
+    public function getReviewKPI(PerformanceReview $performanceReview)
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php:115]: doc comment, return type

File: src/plugins/orangehrmPerformancePlugin/Dto/ReviewEmployeeSupervisorSearchFilterParams.php [hunk 25]
```diff
@@ -0,0 +1,75 @@
...
+namespace OrangeHRM\Performance\Dto;
+
+use OrangeHRM\Core\Dto\FilterParams;
+
+class ReviewEmployeeSupervisorSearchFilterParams extends FilterParams
+{
+    /**
+     * @var int|null
+     */
+    protected ?int $EmpNumber = null;
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Dto/ReviewEmployeeSupervisorSearchFilterParams.php:29]:
```suggestion
    protected ?int $empNumber = null;
```
refactor getters setters as well

File: src/plugins/orangehrmPerformancePlugin/Service/PerformanceReviewService.php [hunk 26]
```diff
@@ -0,0 +1,81 @@
...
+use OrangeHRM\Entity\PerformanceReview;
+use OrangeHRM\Entity\Subunit;
+use OrangeHRM\Performance\Dao\PerformanceReviewDao;
+use OrangeHRM\Performance\Exception\ReviewServiceException;
+
+/**
+ * Description of PerformanceReviewService
+ *
+ * @author nadeera
+ */
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Service/PerformanceReviewService.php:33]:
```suggestion
```

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue (hunk 6 above)
Comment (RajithaKumara) [reply src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue:93]: @SharukaNP seems this is not addressed

File: src/client/src/orangehrmPerformancePlugin/pages/EditReview.vue [hunk 27]
```diff
@@ -0,0 +1,239 @@
...
+          activate,
+        })
+        .then(() => {
+          return this.$toast.saveSuccess();
+        })
+        .catch(error => {
+          return this.$toast.warn({
+            title: this.$t('general.warning'),
+            message: error,
+          });
```
Comment (RajithaKumara) [inline src/client/src/orangehrmPerformancePlugin/pages/EditReview.vue:230]:
```suggestion
        .catch(response => {
          return this.$toast.warn({
            title: this.$t('general.warning'),
            message: response?.data.error.message,
          });
```

File: src/plugins/orangehrmPerformancePlugin/Api/Model/SupervisorModel.php [hunk 28]
```diff
@@ -0,0 +1,50 @@
...
+                ['getSupervisor','getFirstName'],
+                ['getSupervisor','getMiddleName'],
+            ]
+        );
+        $this->setAttributeNames(
+            [
+                ['empNumber'],
+                ['lastName'],
+                ['firstName'],
+                ['middleName'],
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Api/Model/SupervisorModel.php:46]: `terminationId`

File: src/plugins/orangehrmPerformancePlugin/Api/ReviewEmployeeSupervisorAPI.php [hunk 29]
```diff
@@ -0,0 +1,159 @@
...
+{
+    use PerformanceReviewServiceTrait;
+
+    public const FILTER_EMPLOYEE_NUMBER = 'empNumber';
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Api/ReviewEmployeeSupervisorAPI.php:40]:
```suggestion
```

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmPerformancePlugin/Api/PerformanceTrackerAPI.php [hunk 30]
```diff
@@ -220,13 +220,13 @@ public function update(): EndpointResult
         $this->throwRecordNotFoundExceptionIfNotExist($performanceTracker, PerformanceTracker::class);
         $this->setPerformanceTrackerParams($performanceTracker);
         $performanceTracker->setModifiedDate($this->getDateTimeHelper()->getNow());
-        $reviewers = $this->getRequestParams()->getArray(
+        $reviewerEmpNumbers = $this->getRequestParams()->getArray(
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Api/PerformanceTrackerAPI.php:223]: can any employee be a reviewer?

File: src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php [hunk 31]
```diff
@@ -39,22 +41,43 @@ class PerformanceReviewDao extends BaseDao
...
             ->andWhere('rt.subordinate = :empNumber')
             ->setParameter('empNumber', $reviewEmployeeSupervisorSearchFilterParams->getEmpNumber());
         if (! is_null($reviewEmployeeSupervisorSearchFilterParams->getNameOrId())) {
-            $q->andWhere(
-                $q->expr()->orX(
-                    $q->expr()->like('employee.firstName', ':nameOrId'),
-                    $q->expr()->like('employee.lastName', ':nameOrId'),
-                    $q->expr()->like('employee.middleName', ':nameOrId'),
-                    $q->expr()->like('employee.employeeId', ':nameOrId'),
+            $qb->andWhere(
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php:69]: hope you understand the value of unit testing

File: src/plugins/orangehrmPerformancePlugin/entity/Decorator/PerformanceReviewDecorator.php [hunk 32]
```diff
@@ -80,6 +80,7 @@ public function getSupervisorReviewer(): Reviewer
         return array_values($supervisorArray)[0];
     }
 
+
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/entity/Decorator/PerformanceReviewDecorator.php:83]:
```suggestion
```

**Review by RajithaKumara — APPROVED** [review]

File: src/plugins/orangehrmPerformancePlugin/config/PerformancePluginConfiguration.php [hunk 33]
```diff
@@ -18,6 +18,7 @@
  */
 
 use OrangeHRM\Performance\Service\KpiService;
+
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/config/PerformancePluginConfiguration.php:21]:
```suggestion
```

File: src/plugins/orangehrmPerformancePlugin/config/routes.yaml [hunk 34]
```diff
@@ -12,6 +12,34 @@ apiv2_performance_kpis:
   defaults:
     _api: OrangeHRM\Performance\Api\KpiAPI
 
+apiv2_performance_supervisors:
+  path: /api/v2/performance/supervisors
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/config/routes.yaml:16]:
```suggestion
  path: /api/v2/performance/reviews/supervisors
```
:thinking: how about this

File: src/plugins/orangehrmPerformancePlugin/config/routes.yaml [hunk 35]
```diff
@@ -12,6 +12,34 @@ apiv2_performance_kpis:
...
+  methods: [ GET, POST, DELETE ]
+  defaults:
+    _api: OrangeHRM\Performance\Api\ReviewEmployeeSupervisorAPI
+
+apiv2_performance_review:
+  path: /api/v2/performance/reviews/{id}
+  controller: OrangeHRM\Core\Controller\Rest\V2\GenericRestController::handle
+  methods: [ GET, PUT ]
+  defaults:
+    _api: OrangeHRM\Performance\Api\PerformanceReviewAPI
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPerformancePlugin/config/routes.yaml:27]:
```suggestion
apiv2_performance_review:
  path: /api/v2/performance/reviews/{id}
  controller: OrangeHRM\Core\Controller\Rest\V2\GenericRestController::handle
  methods: [ GET, PUT ]
  defaults:
    _api: OrangeHRM\Performance\Api\PerformanceReviewAPI
  requirements:
    empNumber: '\d+'
```

File: src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue [hunk 36]
```diff
@@ -0,0 +1,214 @@
...
+            <oxd-button
+              display-type="ghost"
+              :label="$t('general.cancel')"
+              type="button"
+              @click="onCancel"
+            />
+            <oxd-button
+              display-type="ghost"
+              :label="$t('general.save')"
+              type="submit"
```
Comment (SharukaNP) [inline src/client/src/orangehrmPerformancePlugin/pages/AddReview.vue:93]: Changed to fix https://github.com/orangehrm/orangehrm/pull/1292/files/a0e8e2c6064e60e70e94441187cdfdadee41f935#r881173828 @RajithaKumara

File: src/client/src/orangehrmPerformancePlugin/pages/EditPerformanceTracker.vue [hunk 37]
```diff
@@ -100,15 +99,7 @@ export default {
...
-  setup() {
-    const http = new APIService(
-      window.appGlobal.baseUrl,
-      '/api/v2/performance/config/trackers',
-    );
-    return {
-      http,
-    };
-  },
+
```
Comment (Super-Chama) [inline src/client/src/orangehrmPerformancePlugin/pages/EditPerformanceTracker.vue:102]: why this is removed? you need the APIService to make API calls? I see API calls in this components

File: src/client/src/orangehrmPerformancePlugin/pages/EditReview.vue [hunk 38]
```diff
@@ -0,0 +1,239 @@
...
+    };
+    return {
+      http,
+      rules,
+    };
+  },
+  data() {
+    return {
+      isLoading: false,
+      review: {...reviewModel},
```
Comment (Super-Chama) [inline src/client/src/orangehrmPerformancePlugin/pages/EditReview.vue:173]: you declared review inside `setup()` why re-declare here? add it to return block of `setup()`

Comment (Super-Chama) [reply src/client/src/orangehrmPerformancePlugin/pages/EditReview.vue:173]: also move `isLoading` as a ref also. then can remove `data()`

File: src/client/src/orangehrmPerformancePlugin/pages/ReviewSearch.vue [hunk 39]
```diff
@@ -350,7 +351,9 @@ export default {
           },
         };
       } else if (row.statusName === 'Inactive') {
+        //TODO:: Change to Id
```
Comment (Super-Chama) [inline src/client/src/orangehrmPerformancePlugin/pages/ReviewSearch.vue:354]: why TODO?

**Review by Super-Chama — CHANGES_REQUESTED** [review]

File: src/client/src/orangehrmPerformancePlugin/pages/EditPerformanceTracker.vue (hunk 37 above)
Comment (SharukaNP) [reply src/client/src/orangehrmPerformancePlugin/pages/EditPerformanceTracker.vue:102]: Added it back

File: src/client/src/orangehrmPerformancePlugin/pages/EditReview.vue (hunk 38 above)
Comment (SharukaNP) [reply src/client/src/orangehrmPerformancePlugin/pages/EditReview.vue:173]: I'll move the rules to data to keep it same as AddReview

File: src/client/src/orangehrmPerformancePlugin/pages/ReviewSearch.vue (hunk 39 above)
Comment (SharukaNP) [reply src/client/src/orangehrmPerformancePlugin/pages/ReviewSearch.vue:354]: We are to remove the use of status names and use ids instead. Change to be added in the next PR

**Review by Super-Chama — APPROVED** [review]

**Pull request merged**